docs/docs/03-usage/10-watermarker/03-java.md

```diff
@@ -85,99 +77,178 @@ public class Main {
         // ***** Multiple watermarks *****
         // *******************************
 
-        // for multiple watermarks in a single text the parameter 'singleWatermark' must be set 
-        // to 'false' when passed to the extraction function alongside the watermarked text, 
-        // details are linked above this code block 
+        // for multiple watermarks in a single text the parameter 'singleWatermark' must be set
+        // to 'false' when passed to the extraction function alongside the watermarked text,
+        // details are linked above this code block
 
         // a second Watermark to illustrate multiple watermark extraction
         String secondWatermarkText = "Okay";
 
         // creating the second watermark
-        TextWatermark secondWatermark = TextWatermark.Companion.create(secondWatermarkText);
+        TextWatermark secondWatermark = TextWatermark.raw(secondWatermarkText);
 
         // inserting the second watermark into the coverText
-        String secondWatermarkedText = unwrap(watermarker.textAddWatermark(coverText, secondWatermark));
+        Result<String> secondInsertionResult = watermarker.textAddWatermark(coverText, secondWatermark);
+        String secondWatermarkedText;
+        if (secondInsertionResult.isSuccess() && secondInsertionResult.hasValue()) {
+            System.out.println("watermarked second text:");
+            secondWatermarkedText = secondInsertionResult.getValue();
+            System.out.println(watermarkedText);
+        } else {
+            System.out.println("watermarked text not found");
+            return;
+        }
 
         // combining the watermarked texts to get two different watermarks in one Text
         String combinedText = watermarkedText + secondWatermarkedText;
 
         // extract the watermarks from the watermarked text
-        List<TextWatermark> extractedMultipleWatermarks =
-                unwrap(watermarker.textGetTextWatermarks(combinedText, true, false, false));
+        // notice that we set the 'singleWatermark' argument to 'false' here:
+        Result<List<TextWatermark>> secondExtractionResult = watermarker.textGetTextWatermarks(combinedText, true, false, false);
+        if (secondExtractionResult.isSuccess() && secondExtractionResult.hasValue()) {
+            List<TextWatermark> extractedMultipleWatermarks = secondExtractionResult.getValue();
+            // print the watermarks found
+            extractedMultipleWatermarks.forEach(System.out::println);
+        } else {
+            System.out.println("second watermark extraction failed");
+        }
+        System.out.println();
 
-        // print the watermarks found
-        extractedMultipleWatermarks.forEach(System.out::println);
-    
-    }
+        // also notice that it is generally not recommended to concatenate two watermarked texts. this mainly serves
+        // to demonstrate that the extraction is capable of identifying several different watermark contents.
```

Comment:
Since we are working with a markdown documentation, we can integrate those hints as classical text and create multiple blocks of small coding examples instead of one big code block with commented hints inside.